hpx/plugins/parcel/message_buffer.hpp

```diff
@@ -60,7 +60,27 @@ namespace hpx { namespace plugins { namespace parcel { namespace detail
         void operator()(parcelset::parcelport* pp)
         {
             if (!messages_.empty())
+            {
+                if (0 == threads::get_self_ptr())
```

Comment:
Right... it goes directly through the parcelport... however, I'm wondering if it shouldn't go through the parcelhandler instead, especially if migration is involved, we might need to resolve the destination again.